platforms/ide/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/EclipseClasspath.java

```diff
@@ -195,13 +186,25 @@ public void setSourceSets(Iterable<SourceSet> sourceSets) {
      * See {@link EclipseClasspath} for an example.
      */
     public Collection<Configuration> getPlusConfigurations() {
-        return plusConfigurations;
+        return getPlusConfigurationsProperty().get();
     }
 
     public void setPlusConfigurations(Collection<Configuration> plusConfigurations) {
-        this.plusConfigurations = plusConfigurations;
+        this.getPlusConfigurationsProperty().set(plusConfigurations);
     }
 
+    /**
+     * The configurations whose files are to be added as classpath entries.
+     * <p>
+     * See {@link EclipseClasspath} for an example.
+     *
+     * @return the configurations whose files are to be added as classpath entries
+     *
+     * @since 9.0
+     */
+    @Incubating
+    abstract public ListProperty<Configuration> getPlusConfigurationsProperty();
```

Comment:
❓ This is adding new public API duplicating an existing one with some synthetic name. I would be more inclined we figure out a way that doesn't require new public API that we may want to remove eventually or using bad names. 

Package private visibility for `Property<T>` fields is not possible here because of the package structure, the plugin setting the conventions is in a different package. What about the conventions to be set in this class constructor instead? What about introducing property getter on an internal protocol, like `EclipseClasspathInternal` ?

Reply:
I'm open to any suggestion, but after removing convention mappings at a couple of other places, this approach was seemingly the only way forward.